packages/bns/src/index.ts

```diff
@@ -0,0 +1,748 @@
+import {
+  bufferCV,
+  callReadOnlyFunction,
+  ClarityType,
+  ClarityValue,
+  cvToString,
+  getAddressFromPrivateKey,
+  hash160,
+  makeRandomPrivKey,
+  makeUnsignedContractCall,
+  privateKeyToString,
+  ResponseErrorCV,
+  StacksTransaction,
+  standardPrincipalCV,
+  UnsignedContractCallOptions,
+} from '@stacks/transactions';
+
+import { StacksNetwork } from '@stacks/network';
+
+import { bufferCVFromString, decodeFQN, getZonefileHash, uintCVFromBN } from './utils';
+
+import BN from 'bn.js';
+
+export const BNS_CONTRACT_ADDRESS = 'ST000000000000000000002AMW42H';
+export const BNS_CONTRACT_NAME = 'bns';
+
+export interface PriceFunction {
+  base: BN;
+  coefficient: BN;
+  b1: BN;
+  b2: BN;
+  b3: BN;
+  b4: BN;
+  b5: BN;
+  b6: BN;
+  b7: BN;
+  b8: BN;
+  b9: BN;
+  b10: BN;
+  b11: BN;
+  b12: BN;
+  b13: BN;
+  b14: BN;
+  b15: BN;
+  b16: BN;
+  nonAlphaDiscount: BN;
+  noVowelDiscount: BN;
+}
+
+export interface BnsContractCallOptions {
+  functionName: string;
+  functionArgs: ClarityValue[];
+  publicKey: string;
+  network: StacksNetwork;
+  attachment?: Buffer;
+}
+
+async function makeBnsContractCall(options: BnsContractCallOptions): Promise<StacksTransaction> {
+  const txOptions: UnsignedContractCallOptions = {
+    contractAddress: BNS_CONTRACT_ADDRESS,
+    contractName: BNS_CONTRACT_NAME,
+    functionName: options.functionName,
+    functionArgs: options.functionArgs,
+    publicKey: options.publicKey,
+    validateWithAbi: false,
+    network: options.network,
+  };
+
+  return makeUnsignedContractCall(txOptions);
+}
+
+export interface BnsReadOnlyOptions {
+  functionName: string;
+  functionArgs: ClarityValue[];
+  senderAddress: string;
+  network: StacksNetwork;
+}
+
+async function callReadOnlyBnsFunction(options: BnsReadOnlyOptions): Promise<ClarityValue> {
+  return callReadOnlyFunction({
+    contractAddress: BNS_CONTRACT_ADDRESS,
+    contractName: BNS_CONTRACT_NAME,
+    functionName: options.functionName,
+    senderAddress: options.senderAddress,
+    functionArgs: options.functionArgs,
+    network: options.network,
+  });
+}
+
+/**
+ * Check if name can be registered
+ *
+ * @param {string} fullyQualifiedName - the fully qualified name to check
+ * @param {StacksNetwork} network - the Stacks network to broadcast transaction to
+ *
+ * @returns {Promise} that resolves to true if the operation succeeds
+ */
+export async function canRegisterName(
+  fullyQualifiedName: string,
+  network: StacksNetwork
+): Promise<boolean> {
```

Comment:
What are you thoughts on using an object here for args? Makes it a bit cleaner if the function args need expanded in the future

Reply:
Not opposed. Do you feel this way about the other similar read only functions too?

Reply:
I think generally functions with two or more args should use an object, since the trend is for the amount of options to increase.